openedx/openedx-platform
# Use django default storage for ORA2 file uploads

Updates `lms.env` settings and `lms.urls` to allow ORA2 file uploads to work with the default django storage configured for the platform.

* Updates `lms.envs.openstack` and `lms.envs.devstack` to use django storage for ORA2.
* Adds the ORA2 file upload endpoints to `lms.urls` under `/openassessment/fileupload/`
* Enables `SWIFT_CONTENT_TYPE_FROM_FD` by default in `lms.envs.openstack` settings, which allows the django swift storage backend to automatically determine the mimetype of uploaded files, so that their `Content-Type` header is correct when downloaded.

**JIRA tickets**: [OSPR-1809](https://openedx.atlassian.net/browse/OSPR-1809)

**Dependencies**: https://github.com/edx/edx-ora2/pull/1018

**Sandbox URL**:

* LMS: https://pr15446.sandbox.opencraft.hosting/
* Studio: https://studio-pr15446.sandbox.opencraft.hosting/

**Partner information**: 3rd party-hosted open edX instance

**Deployment targets**: edx.org and edge.edx.org

However, since the `lms.env.aws` settings file remains untouched, and no change was made to the ORA2 default storage mechanism (`s3`), this change will not affect AWS deployments like edx.org.

**Merge deadline**: None

**Testing instructions**:

See https://github.com/edx/edx-ora2/pull/1018 for devstack and openstack sandbox testing instructions.

**Author notes and concerns**:

1. *TODO* Update ORA2 requirement to use newly tagged version once 
https://github.com/edx/edx-ora2/pull/1018 merges.
1. I chose not to submit a configuration PR to set `ORA2_FILEUPLOAD_BACKEND` and `SWIFT_CONTENT_TYPE_FROM_FD`, but could if required.  This PR was sufficient for our needs.

**Reviewers**
- [ ] edX reviewer[s] TBD - CC @gsong
